src/lib/lit-extended.ts

```diff
@@ -144,17 +144,22 @@ export class EventPart implements Part {
 
   setValue(value: any): void {
     const listener = getValue(this, value);
-    const previous = this._listener;
-    if (listener === previous) {
+    if (listener === this._listener) {
       return;
     }
-
-    this._listener = listener;
-    if (previous != null) {
-      this.element.removeEventListener(this.eventName, previous);
+    if (listener == null) {
+      this.element.removeEventListener(this.eventName, this);
+    } else if (this._listener == null) {
+      this.element.addEventListener(this.eventName, this);
     }
-    if (listener != null) {
-      this.element.addEventListener(this.eventName, listener);
+    this._listener = listener;
+  }
+
+  handleEvent(event: Event) {
+    if (typeof this._listener === 'function') {
```

Comment:
It's possible that the listener could be an object with a `handleEvent` method. I see I didn't handle this in my previous implementation either...